crates/typst/src/eval/mod.rs

```diff
@@ -399,9 +399,81 @@ fn eval_markup<'a>(
     let flow = vm.flow.take();
     let mut seq = Vec::with_capacity(exprs.size_hint().1.unwrap_or_default());
 
+    // That stores previous text node
+    // to merge it with next adjacent one if present.
+    let mut previous_text: Option<EcoString> = None;
+
+    // Let's believe that two adjacent spaces is gibberish.
+    // That variable helps to join two texts, if they are separated by spaces only
+    // And move space to it's own node otherwise.
+    let mut stored_space: Option<ast::Expr<'_>> = None;
+
+    // Unfortunately, it is currently impossible to attach exact span to merged text node,
+    // So let's attach at least some of them.
+    let mut last_span: Span = Span::detached();
```

Comment:
I hope my question isn't too ignorant, but is it possible to perhaps add joining functionality to Span to make this possible? I'm sure people would see it as a bug if Typst only highlighted up to the first escape of some error text or something (although that wouldn't be the end of the world, but worth considering).

Reply:
Again, it's a design problem. Spans store their id-s that are created during parsing syntax nodes. Spans of children should be less than span of parent, and spans of next children should be larger than span of previous, something like that. Each expression stores &SyntaxNode that has it's own span.

I'm not sure, of course, but it seems to me that creating new spans dynamically during evaluation just isn't possible without changing how it works.